# Apps With More Users

## - This is a project to help improve the company's revenue by analysing data of users who use our app.

## - Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

## Opening and exploring the data

In [1]:
from csv import reader

opened_AppleStore = open('AppleStore.csv')
read_AppleStore = reader(opened_AppleStore)
apps_AppleStore = list(read_AppleStore)
ios_head = apps_AppleStore[0]
ios= apps_AppleStore[1:]

opened_googleplaystore = open('googleplaystore.csv')
read_googleplaystore = reader(opened_googleplaystore)
apps_googleplaystore = list(read_googleplaystore)
google_head = apps_googleplaystore[0]
google = apps_googleplaystore[1:]



In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        

In [3]:
print(ios_head)
print('\n')
explore_data(apps_AppleStore,1,4,rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7198
Number of columns: 16


In [4]:
print(google_head)
print('\n')
explore_data(apps_googleplaystore,1,4,rows_and_columns=True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10842
Number of columns: 13


## Deleting wrong data

In [5]:
# To find any missing values in any rows in the dataset

for row in ios:
    if len(row) != len(ios_head):
        print(row)
    else:
        print('Dataset is ok at this point')
        break

Dataset is ok at this point


In [6]:
import pandas as pd
f = pd.read_csv('googleplaystore.csv')
f[10471:10474]

App         Category  Rating  \
10471                         Xposed Wi-Fi-Pwd  PERSONALIZATION     3.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   
10473                  osmino Wi-Fi: free WiFi            TOOLS     4.2   

      Reviews    Size     Installs  Type     Price Content Rating  \
10471    1042    404k     100,000+  Free         0       Everyone   
10472    3.0M  1,000+         Free     0  Everyone            NaN   
10473  134203    4.1M  10,000,000+  Free         0       Everyone   

                  Genres    Last Updated Current Ver   Android Ver  
10471    Personalization  August 5, 2014       3.0.0  4.0.3 and up  
10472  February 11, 2018          1.0.19  4.0 and up           NaN  
10473              Tools  August 7, 2018     6.06.14    4.4 and up

In [7]:
f.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [8]:
# print(ios[10472])
print(google[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [9]:
print(len(google))

10841


In [10]:
del google[10472]

In [11]:
print(len(google))

10840


## Removing Duplicate Entries

In [12]:
for app in google:
    app_name = app[0]
    if app_name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [13]:
duplicate_apps = []
unique_apps = []

for app in google:
    app_name = app[0]
    if app_name in unique_apps:
        duplicate_apps.append(app_name)
    else:
        unique_apps.append(app_name)

print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [14]:
for i in unique_apps[:15]:
    print(i)

Photo Editor & Candy Camera & Grid & ScrapBook
Coloring book moana
U Launcher Lite – FREE Live Cool Themes, Hide Apps
Sketch - Draw & Paint
Pixel Draw - Number Art Coloring Book
Paper flowers instructions
Smoke Effect Photo Maker - Smoke Editor
Infinite Painter
Garden Coloring Book
Kids Paint Free - Drawing Fun
Text on Photo - Fonteee
Name Art Photo Editor - Focus n Filters
Tattoo Name On My Photo Editor
Mandala Coloring Book
3D Color Pixel by Number - Sandbox Art Coloring


In [15]:
for i in duplicate_apps[:15]:
    print(i)

Quick PDF Scanner + OCR FREE
Box
Google My Business
ZOOM Cloud Meetings
join.me - Simple Meetings
Box
Zenefits
Google Ads
Google My Business
Slack
FreshBooks Classic
Insightly CRM
QuickBooks Accounting: Invoicing & Expenses
HipChat - Chat Built for Teams
Xero Accounting Software


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows we printed two cells above for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was collected at different times. We can use this to build a criterion for keeping rows. We won't remove rows randomly, but rather we'll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings.

To do that, we will:

Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
Use the dictionary to create a new data set, which will have only one entry per app (and we only select the apps with the highest number of reviews)

In [16]:
# If duplicated apps were removed, we'd be left with
print(len(google) - len(duplicate_apps), 'apps left')

9659 apps left


In [17]:
reviews_max = {}
for app_name in google:
    name = app_name[0]
    n_reviews = float(app_name[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))
print('\n')
# print(reviews_max)

9659




Expected length: 9659

Actual length: 9659

Now, let's use the reviews_max dictionary to remove the duplicates. For the duplicate cases, we'll only keep the entries with the highest number of reviews. In the code cell below:

We start by initializing two empty lists, android_clean and already_added.
We loop through the android data set, and for every iteration:
We isolate the name of the app and the number of reviews.
We add the current row (app) to the android_clean list, and the app name (name) to the already_cleaned list if:
The number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary; and
The name of the app is not already in the already_added list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for reviews_max[name] == n_reviews, we'll still end up with duplicate entries for some apps.

In [18]:
android_clean  = []
already_added = []

for app_name in google:
    name = app_name[0]
    n_reviews = float(app_name[3])
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app_name)
        already_added.append(name)

Exploring our new dataset

In [19]:
explore_data(android_clean, 0, 4, rows_and_columns=True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 9659
Number of columns: 13


Number of rows tally as expected

## Removing Non-English Apps

Basically, we are concerned with apps that are targeted towards English-speaking audience and as a result remove all other apps that are otherwise

Checking this rows, we see examples of apps that are not targeted to English-speaking audience

In [20]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We built this function below, and we use the built-in ord() function to find out the corresponding encoding number of each character.

In [21]:
def english_or_not(string):
    for i in string:
        if ord(i) > 127:
            return False
    return True
english_or_not('Docs To Go™ Free Office Suite')

False

In [22]:
'''for character in google:
    for non in character:
        for i in non:
            if ord(i) > 127:
                print(character,'\n')
'''

"for character in google:\n    for non in character:\n        for i in non:\n            if ord(i) > 127:\n                print(character,'\n')\n"

In [23]:
def english_or_not(string):
    non_ascii = 0
    for i in string:
        if ord(i) > 127:
            non_ascii += 1
    if non_ascii <= 3:
        return True
    else:
        return False
print(english_or_not('Docs To Go™ Free Office Suite'))
print(english_or_not('Instachat 😜'))
print(english_or_not('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


Let's filter out the non-English apps from both data sets - using the english_or_not() function

In [24]:
android_English = []
ios_English = []

for app in android_clean:
    name = app[0]
    if english_or_not(name):
        android_English.append(app)
        
for app in ios:
    name = app[1]
    if english_or_not(name):
        ios_English.append(app)
        
explore_data(android_English,0,3,rows_and_columns=True)
print('\n')
explore_data(ios_English,0,3,rows_and_columns=True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

## Isolating Free Apps

In [25]:
android_free_apps = []
ios_free_apps = []

for app in android_English:
    price = app[7]
    if price == '0':
        android_free_apps.append(app)

for app in ios_English:
    price = app[4]
    if price == '0.0':
        ios_free_apps.append(app)
        
explore_data(android_free_apps,0,3,rows_and_columns=True)
print('\n')
explore_data(ios_free_apps,0,3,rows_and_columns=True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

This shows we are left to having 8864 (free) android apps and 3222 (free) ios apps to work with

## Most common apps by Genre

In [26]:
def freq_table(dataset,index):
    freq_tab = {}
    total = 0
    
    for row in dataset:
        total += 1
        data = row[index]
        if data in freq_tab:
            freq_tab[data] += 1
        else:
            freq_tab[data] = 1
    
    percent_freq_tab = {}
    for key in freq_tab:
        percentage = (freq_tab[key] / total) * 100
        percent_freq_tab[key] = percentage
        
    return percent_freq_tab

In [27]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

## Using display_table() function to display the frequency table of columns Prime_genre, Genres, and Category

In [28]:
#Frequency table for prime_genre column of the App Store data set
display_table(ios_English, 11)

Games : 54.860100274947435
Entertainment : 7.261846999838266
Education : 6.6310852337053205
Photo & Video : 5.515122109008572
Utilities : 3.4449296458030085
Productivity : 2.7171276079573023
Health & Fitness : 2.6686074721009216
Music : 2.215752870774705
Social Networking : 2.037845705967977
Sports : 1.6820313763545207
Lifestyle : 1.6011644832605532
Shopping : 1.3747371825974446
Weather : 1.1159631246967492
Travel : 0.9704027171276078
News : 0.9218825812712276
Book : 0.8895358240336406
Reference : 0.8571890667960537
Business : 0.8571890667960537
Finance : 0.7924955523208799
Food & Drink : 0.7116286592269124
Navigation : 0.452854601326217
Medical : 0.3396409509946628
Catalogs : 0.08086689309396733


We find out that the free English apps(Apple apps) are dominated/populated with GAMES with a percentage of about 54%, while Entertainment, Education and Photo & Video have about 7%, 6% and 5.5% respectively. However, note that the population or number of apps doesn't determine the demand of the app.

In [29]:
#Frequency table for Genre column of the App Store data set
display_table(android_free_apps, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Here, unlike the Apple app, we see a more balanced distribution of Google app distribution/population where TOOL,ENTERTAINMENT and EDUCATION apps are the first 3 on the list.

In [30]:
#Frequency table for Category column of the App Store data set
display_table(android_free_apps, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The Family category for Google apps which amounts to about 19% indicates that the it is mostly games for kids.

## Most popular apps by genre on the App store

In [31]:
genre_ios = freq_table(ios_English,-5)

for genre in genre_ios:
    total = 0
    len_genre = 0
    for app in ios_English:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre,':',avg_n_ratings)

Food & Drink : 19934.386363636364
Shopping : 26635.011764705883
Utilities : 7927.525821596244
Reference : 27037.188679245282
Catalogs : 3465.0
Sports : 15350.913461538461
Social Networking : 60253.84920634921
Education : 2472.278048780488
Medical : 648.952380952381
Photo & Video : 14688.715542521993
Entertainment : 8862.409799554565
Book : 10359.2
News : 16980.315789473683
Navigation : 19370.821428571428
Music : 29047.109489051094
Health & Fitness : 10802.157575757576
Lifestyle : 8930.373737373737
Games : 15586.759433962265
Weather : 23145.246376811596
Business : 5149.320754716981
Travel : 19030.183333333334
Finance : 23353.530612244896
Productivity : 8508.089285714286


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [32]:
for app in ios_English:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
MotionX GPS : 14970
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
Gaia GPS Classic : 2429
Plane Finder - Flight Tracker : 1438
iMaps+ for Google Maps ™ and Street View ™ : Transit and Offline Contacts : 1225
NAVIGON Europe : 927
Localscope - Find places and people around you : 868
Ski Tracks : 829
TRANSPORT MODS for MINECRAFT Pc EDITION : 754
Pocket Earth PRO Offline Maps & Travel Guides : 748
Ship Finder : 624
Boating USA : 342
Maps 3D PRO - GPS for Bike, Hike, Ski & Outdoor : 280
Cachly - Simple and powerful Geocaching for iPhone : 263
ImmobilienScout24: Real Estate Search in Germany : 187
The JMU Bus App : 35
Avertinoo : 32
iStellar : 30
mySTATE - State College : 26
Road watcher: dash camera, car video recorder. : 10
Streets – Street View Browser : 10
Railway Route Search : 5
parkOmator – for Apple Watch meter expiration timer, notifications & GPS navigator t

In [33]:
for app in ios_English:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Sky Guide: View Stars Night or Day : 22100
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
Dictionary.com Dictionary & Thesaurus Premium : 11530
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
WolframAlpha : 7410
e-Sword HD: Bible Study Made Easy : 7309
iHandy Translator Pro : 5163
Dictionary.com Premium Dictionary & Thesaurus for iPad : 4922
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Speak & Translate － Live Voice and Text Translator : 4344
National Geographic World Atlas : 4255
Knots 3D : 3196
iQuran : 2929
Merriam-Webster Dictionary & Thesaurus : 2843
e-Sword LT: Bible Study on the Go : 2152
GUNS MODS for Minecraf

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

## Most popular apps by genre on the Google Play

In [34]:
display_table(android_English, 5)

1,000,000+ : 14.70771791137924
100,000+ : 11.504056584148117
10,000+ : 10.619929269814854
10,000,000+ : 9.746203453297275
1,000+ : 9.153318077803203
100+ : 7.322654462242563
5,000,000+ : 6.292906178489703
500,000+ : 5.242354899105471
5,000+ : 4.836696484293738
50,000+ : 4.815893488662367
10+ : 3.994175161223216
500+ : 3.4116912835448305
50,000,000+ : 2.1219055543998335
50+ : 2.1219055543998335
100,000,000+ : 1.9658830871645518
5+ : 0.8529228208862076
1+ : 0.6864988558352403
500,000,000+ : 0.249635947576451
1,000,000,000+ : 0.20802995631370919
0+ : 0.13521947160391096
0 : 0.010401497815685458


Observing from the above results, the data type presented on the "Installs" column is simply a string as it has commas and plus "+" signs on each row.
Since we don't really need the range of values like the plus sign indicates, we only need to find out which app genre attract the most users, therefore removing the plus sign is no harm to our analysis.

In [35]:
genre_android = freq_table(android_English,1)

for category in genre_android:
    total = 0
    len_category = 0
    
    for app  in android_English:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            n_installs = float(n_installs)
            total += n_installs
            len_category += 1
    avg_n_installs = total / len_category
    print(category,':',avg_n_installs)

SHOPPING : 6966908.880597015
TRAVEL_AND_LOCAL : 13218662.767123288
PARENTING : 525351.8333333334
ART_AND_DESIGN : 1887285.0
MEDICAL : 96944.49873417722
ENTERTAINMENT : 11375402.298850575
GAME : 14256217.600635594
EVENTS : 249580.640625
HEALTH_AND_FITNESS : 3972300.388888889
HOUSE_AND_HOME : 1331540.5616438356
PERSONALIZATION : 4086652.4853333333
BOOKS_AND_REFERENCE : 7641777.871559633
AUTO_AND_VEHICLES : 632501.3214285715
SPORTS : 3373767.6861538463
EDUCATION : 1782566.0377358492
FAMILY : 3345018.516684607
COMMUNICATION : 35153714.17515924
FINANCE : 1319851.4028985507
NEWS_AND_MAGAZINES : 9472807.04
SOCIAL : 22961790.384937238
BEAUTY : 513151.88679245283
BUSINESS : 1663758.627684964
PHOTOGRAPHY : 16636241.267857144
MAPS_AND_NAVIGATION : 3900634.7286821706
PRODUCTIVITY : 15530942.008042896
FOOD_AND_DRINK : 1891060.2767857143
TOOLS : 9785955.211352658
VIDEO_PLAYERS : 24121489.079754602
WEATHER : 4570892.658227848
LIFESTYLE : 1369954.7774725275
COMICS : 817657.2727272727
DATING : 828971.2

Looking at the result above, we see that 'COMMUNICATION' and 'SOCIAL' having 35153714 and 22961790 respectively.

## Let's check for communication and social apps with over 100 million installs.

### For communication apps

In [36]:
for app in android_English:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

### For social apps

In [40]:
for app in android_English:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


If all the communication and social apps that have over 100 million installs are taken out, the average would be reduced roughly ten times

### Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.